File: IsFibo.c
#include <stdio.h>
int main() {    
    int n;
    int isFib=0;
    int i,j;
    int num;
    int fib1=0;
    int fib2=1;
    int nextFib=0;
    scanf("%d",&n);
    while (n!=0) {
        scanf("%d",&num);
        while (nextFib<num) {
            nextFib=fib1+fib2;
            if (nextFib==num) {
                isFib=1;
            }
            fib1=fib2;
            fib2=nextFib;
        }
        if (isFib==1) {
            printf("IsFibo\n");
        }
        else {
            printf("IsNotFibo\n");
        }
        fib1=0;
        fib2=1;
        nextFib=0;
        isFib=0;
        n--;
    }
    return 0;
}

File: WPP_ASS3/IsFibo.py
T=int(input())
result=[]
while T>0 :
    N=int(input())
    fib1=0
    fib2=1
    nextfib=0
    isFib=0
    if N==0 :
        isFib=1
    while nextfib<N :
        nextfib=fib1+fib2
        fib1=fib2
        fib2=nextfib
    if nextfib==N :
        isFib=1
    if isFib==1 :
        result.append("IsFibo")
    else :
        result.append("IsNotFibo")
    T-=1
for i in result :
    print(i)